File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        
        ListNode curr = head;
        int cnt = 0;
        while(curr != null)
        {
            cnt++;
            curr = curr.next;
        }

        int j = cnt-n;

        if(j == 0) return head.next;

        int i = 1;
        curr = head;
        while(curr != null)
        {
            if(i == j)
            {
                curr.next = curr.next.next;
                break;
            }
            i++;
            curr = curr.next;
        }

        return head;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        
        ListNode* dummyHead = new ListNode();
        ListNode* curr = dummyHead;
        int carry = 0;
        
        while(l1 || l2 || carry)
        {
            int v1 = l1 ? l1->val : 0;
            int v2 = l2 ? l2->val : 0;
            int val = v1+v2+carry;
            curr->next = new ListNode(val%10);
            carry = val/10;

            curr = curr->next;
            l1 = l1 ? l1->next : nullptr;
            l2 = l2 ? l2->next : nullptr;
        }
        ListNode* res = dummyHead->next;
        delete dummyHead;
        return res;
    }
};
